### PR TITLE
add initial gitlab ci module

### LINKED ISSUE #423: CI: GitLab CI
Similar to https://github.com/jhipster/jhipster-lite/issues/422

### PR DESCRIPTION
**Pull request by atomfrede**
A very simple initial gitlab ci configuration. Example run/project can be [found here on gitlab](https://gitlab.com/atomfrede/jh-lite-sample-ci/-/jobs/2789026169).

The build file must not be a mustache template, but I guess it needs to be in future iterations maybe.
closes #423

### CODE REVIEW

File: src/main/resources/generator/ci/gitlab/.gitlab-ci-maven.yml.mustache [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  paths:
+    - .m2/repository
+
+stages:
+  - test
+
+Test:run-tests:
+  stage: test
+  script:
+    - chmod +x mvnw
```
Comment (DamnClin) [inline src/main/resources/generator/ci/gitlab/.gitlab-ci-maven.yml.mustache:24]: Not sure this is needed, the init plugin is doing it already, is there a bug here?

Comment (atomfrede) [reply src/main/resources/generator/ci/gitlab/.gitlab-ci-maven.yml.mustache:24]: Will check. Just did the same commands as in github actions, as I thought it is needed.

**Comment by DamnClin** [comment]
Looks good to me but @pascalgrimaud is supposed to be on holiday without PC so I can merge it but can't help for the bounty here :)

File: src/main/resources/generator/ci/gitlab/.gitlab-ci-maven.yml.mustache (hunk 1 above)
Comment (DamnClin) [reply src/main/resources/generator/ci/gitlab/.gitlab-ci-maven.yml.mustache:24]: It may not bee needed in github actions either :)

**Comment by DamnClin** [comment]
Oh, the feature file is missing (to make an actual call to the module WebService). The build will fail for missing coverrage

File: src/main/resources/generator/ci/gitlab/.gitlab-ci-maven.yml.mustache (hunk 1 above)
Comment (atomfrede) [reply src/main/resources/generator/ci/gitlab/.gitlab-ci-maven.yml.mustache]: It is not 😄

**Comment by atomfrede** [comment]
> Looks good to me but @pascalgrimaud is supposed to be on holiday without PC so I can merge it but can't help for the bounty here :)

No worries, no need to hurry.

**Comment by atomfrede** [comment]
> Oh, the feature file is missing (to make an actual call to the module WebService). The build will fail for missing coverrage

Will have a look. Still need to learn how the ci is working.

**Comment by DamnClin** [comment]
It will probably be really close to https://github.com/jhipster/jhipster-lite/blob/main/src/test/features/github-actions.feature

Those files are run by cucumber (you can start CucumberTest as any JUnit test) and are here to make a full round call (so really calling the exposed WebServices and going through the orchestration).

**Comment by atomfrede** [comment]
Bounty claimed https://opencollective.com/generator-jhipster/expenses/88237

**Comment by pascalgrimaud** [comment]
@atomfrede : approved :-)

**Pull request merged**